### PR TITLE
#156008359 implement share functionality

### PR DESCRIPTION
**Pull request by johngorithm**
#### What does this PR do?
- Implement share functionality

#### Description of tasks to be completed

- `cherry-picked` this [commit](https://github.com/johngorithm/github-ap-with-java/pull/14/commits/4ca362b327cab3bdc8a6fcf4c875a08a5e61dc17)
- add share icon to the app tool bar
- launch a share `Intent` on clicking the share icon

#### How should this be manually tested?
- clone this repository
- open the project directory using `Android Studio`
- checkout into `ft-implement-share-functionality-156008359` from Android studio integrated command line
- have an `emulator` downloaded and setup in your IDE
- run the app with `Run` button provided by the IDE and select an emulator
- view a user's profile by clicking on the user's card
- click on the `share icon` display on the toolbar
- `share intent` will be launched with possible means of sharing a user's profile url

#### Any background context you want to provide?
No

#### Screenshots
screen shot 2018-12-16 at 10 03 11 pm

After click on the `share icon`
screen shot 2018-12-16 at 10 03 37 pm

After selecting an option
screen shot 2018-12-16 at 10 04 00 pm

#### What are the relevant Pivotal Tracker Stories?
[156008359](https://www.pivotaltracker.com/story/show/156008359)

### CODE REVIEW

**Comment by petermwash** [comment]
@johngorithm The content share on the PT story description should be `Check out this awesome developer @<github username>, <github profile url>`. I would suggest that you don't change it.

File: app/src/main/java/com/jxw/git_hub_users/view/DetailViewActivity.java [hunk 1]
```diff
@@ -15,7 +32,116 @@ protected void onCreate(Bundle savedInstanceState) {
...
+    }
+
+    @Override
+    public void handleError() {
+        progressDialog.dismiss();
+        Toast.makeText(this, "Something Went Wrong. Please, Try Again", Toast.LENGTH_SHORT).show();
+    }
+
+    @Override
+    public void displayUserProfile(UserProfile userProfile) {
```
Comment (petermwash) [inline app/src/main/java/com/jxw/git_hub_users/view/DetailViewActivity.java:90]: Kindly refactor this method to have at most 25 lines of code.

**Review by petermwash — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/jxw/git_hub_users/view/DetailViewActivity.java (hunk 1 above)
Comment (johngorithm) [reply app/src/main/java/com/jxw/git_hub_users/view/DetailViewActivity.java:90]: Alright ... I will do just that

Comment (johngorithm) [reply app/src/main/java/com/jxw/git_hub_users/view/DetailViewActivity.java:90]: Hi @petermwash 
I have fixed all the requested changes with this *PR*

**Review by petermwash — APPROVED** [review]
LGTM. Good job here.

**Pull request merged**